### ISSUE TITLE
Understand the build process of the source repo (for SAM)

### ISSUE THREAD

**Issue by dorchard** [m2lines/convection-parameterization-in-CAM#3]
This may not be strictly necessary, but would probably be useful from the point of view of understanding the dependencies.

**Comment by jatkinson1000** [comment]
Reopening based on discussions associated with trying to run on CSD3 for purposes of testing refactored parameterisation code.

**Comment by jatkinson1000** [comment]
It would be useful to have SAM running with the old parameterisation and then switch for our refactored version to ensure everything still works as expected before moving to CAM (#ToBeOpened).

I have managed to compile the SAM code provided [here](https://github.com/yaniyuval/Neural_nework_parameterization/tree/main/sam_code_NN) on CSD3 after much effort (build scripts and makefiles etc. were not provided). However, I’m unable to run as there are no testcases or run instructions provided. Based on the documentation for the latest version of SAM I have created a `CaseName` file and copied the `prm` file from `sam_code_NN/sam_cases/run_files_x8_5_layers/` to a new case, but it suggests that I also need, at minimum, an `snd` file, and possibly others such as `grd`, `lsf`, `sfc` etc. It also appears that the case is run from a `RESTART` which has not been provided. I have requested these files and instructions from the authors.
